core/src/avm2/globals/xml.rs

```diff
@@ -600,14 +599,22 @@ pub fn insert_child_after<'gc>(
     }
 
     // 3. Else if Type(child1) is XML
-    if let Some(child1) = child1.and_then(|x| x.as_xml_object()) {
+    if let Some(child1) = child1.as_object().and_then(|x| {
+        if let Some(xml) = x.as_xml_object() {
+            return Some(*xml.node());
+        } else if let Some(list) = x.as_xml_list_object() {
```

Comment:
Please add a note that this is non-standard avmplus behavior.

Reply:
Added.